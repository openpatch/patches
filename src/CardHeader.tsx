import { ReactNode } from "react";
import { Box } from "./Box";
import { Heading, HeadingProps } from "./Heading";
import { HeadingLink } from "./HeadingLink";

export type CardHeaderProps = {
  action?: ReactNode;
  avatar?: ReactNode;
  subtitle?: string;
} & HeadingProps;

export const CardHeader = ({
  action,
  avatar,
  children,
  subtitle,
  ...headingProps
}: CardHeaderProps) => {
  let HeadingComp: any = Heading;
  if (headingProps.href) {
    HeadingComp = HeadingLink;
  }
  return (
    <Box
      display="flex"
      alignItems="center"
      paddingX="standard"
      paddingY="small"
    >
      {avatar && (
        <Box display="flex" alignItems="center">
          {avatar}
        </Box>
      )}
      <Box flex="1" ml={avatar ? "standard" : "none"}>
        <HeadingComp margin="none" fontSize="large" {...headingProps}>
          {children}
        </HeadingComp>
        {subtitle && (
          <Box fontSize="small" as="span">
            {subtitle}
          </Box>
        )}
      </Box>
      {action && (
        <Box display="flex" alignItems="center">
          {action}
        </Box>
      )}
    </Box>
  );
};
