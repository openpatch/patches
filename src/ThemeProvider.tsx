import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
