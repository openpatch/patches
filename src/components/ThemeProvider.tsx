import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
