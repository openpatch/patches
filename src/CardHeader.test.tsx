import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "./ThemeProvider";
import { Default } from "./CardHeader.stories";

const renderWithTheme = () =>
  render(
    <ThemeProvider>
      <Default>Header</Default>
    </ThemeProvider>
  );

test("should render", () => {
  renderWithTheme();
});

test("should match snapshot", () => {
  const { container } = renderWithTheme();
  expect(container).toMatchSnapshot();
});
