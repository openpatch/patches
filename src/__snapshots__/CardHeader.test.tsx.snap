// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 16px;
  padding-bottom: 16px;
}

.emotion-2 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-left: 0px;
}

.emotion-4 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  margin: 0px;
  font-size: 20px;
}

<div>
  <div
    class="emotion-0 emotion-1"
    display="flex"
  >
    <div
      class="emotion-2 emotion-1"
    >
      <h1
        class="emotion-4 emotion-1"
        font-size="large"
      >
        Card Header
      </h1>
    </div>
  </div>
</div>
`;
